persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -218,30 +216,75 @@ abstract class PersistentEntity[Command, Event, State] {
       _state = state
     }
 
+    def setEventHandler[A <: Event](eventClass: Class[A], handler: JFunction[A, State]): Unit =
+      setEventHandlerChangingBehavior[A](eventClass, new JFunction[A, Behavior] {
+        override def apply(evt: A): Behavior = behavior.withState(handler.apply(evt))
+      })
+
+    def setEventHandlerChangingBehavior[A <: Event](eventClass: Class[A], handler: JFunction[A, Behavior]): Unit =
+      eventHandlers = eventHandlers.updated(eventClass, handler)
+
+    def removeEventHandler(eventClass: Class[_ <: Event]): Unit =
+      eventHandlers -= eventClass
+
+    def setCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiFunction[A, CommandContext[R], Persist[_ <: Event]]
+    ): Unit = {
+      commandHandlers = commandHandlers.updated(
+        commandClass,
+        handler.asInstanceOf[JBiFunction[A, CommandContext[Any], Persist[_ <: Event]]]
+      )
+    }
+
+    def removeCommandHandler(commandClass: Class[_ <: Command]): Unit =
+      commandHandlers -= commandClass
+
+    def setReadOnlyCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiConsumer[A, ReadOnlyCommandContext[R]]
+    ): Unit = {
+      setCommandHandler[R, A](commandClass, new JBiFunction[A, CommandContext[R], Persist[_ <: Event]] {
+        override def apply(cmd: A, ctx: CommandContext[R]): Persist[Event] = {
+          handler.accept(cmd, ctx)
+          ctx.done()
+        }
+      });
+    }
+
+    def build(): Behavior =
+      Behavior(_state, eventHandlers, commandHandlers)
+
+  }
+
+  /**
+   * Mutable builder that is used for defining the event and command handlers.
+   * Use [[BehaviorBuilder#build]] to create the immutable [[PersistentEntity.Behavior]].
+   */
```

Comment:
(Restoring a comment that became "outdated" when I pushed a change to the branch)

An example of use here would be good (could take code from the docs). Also a good idea to link back to the documentation page that describes it in detail.